src/DetailView.php

```diff
@@ -371,23 +372,25 @@ private function renderFields(): string
         $rows = [];
 
         foreach ($fields as $field) {
-            $label = strtr($this->labelTemplate, [
-                '{label}' => $field['label'],
-                '{tag}' => $field['labelTag'],
-                '{attributes}' => Html::renderTagAttributes($field['labelAttributes']),
-            ]);
-
-            $value = strtr($this->valueTemplate, [
-                '{value}' => $field['value'],
-                '{tag}' => $field['valueTag'],
-                '{attributes}' => Html::renderTagAttributes($field['valueAttributes']),
-            ]);
-
-            $rows[] = strtr($this->fieldTemplate, [
-                '{attributes}' => Html::renderTagAttributes($this->fieldAttributes),
-                '{label}' => $label,
-                '{value}' => $value,
-            ]);
+            if ($field['isVisible']) {
```

Comment:
I'd skip the iteration instead of wrapping code. Reads a bit better.

```suggestion
            if (!$field['isVisible']) {
                continue;
            }
```